### PR TITLE
[Stable blocking] Prevent showing screenShareButton when formFactor is set to mobile

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What
Prevent showing screenShareButton when formFactor is set to mobile

# Why
CallControlOptions `screenShareButton` was set to true by accident on sample app for all devices that are not iOS. We should prevent showing the screenShareButton when formFactor is set to mobile

# How Tested
Local testing of Calling and CallwithChat samples on Desktop and Android

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: samples/CallWithChat/src/app/views/CallScreen.tsx [hunk 1]
```diff
@@ -150,9 +150,11 @@ export const CallScreen = (props: CallScreenProps): JSX.Element => {
 
   const options: CallWithChatCompositeOptions = useMemo(
     () => ({
-      callControls: {
-        screenShareButton: !shouldDisableScreenShare
```
Comment (JamesBurnside) [inline samples/CallWithChat/src/app/views/CallScreen.tsx:154]: Is this change necessary anymore if we are dictating never show screenshare on mobile?

File: samples/CallWithChat/src/app/views/CallScreen.tsx [hunk 2]
```diff
@@ -150,9 +150,11 @@ export const CallScreen = (props: CallScreenProps): JSX.Element => {
...
-      }
+      callControls: shouldDisableScreenShare
```
[the remark this answers is not in the document]
Comment (JamesBurnside) [reply samples/CallWithChat/src/app/views/CallScreen.tsx:153]: (Or fix the shouldDisableScreenshare to be isMobile instead of isIOS)

File: samples/CallWithChat/src/app/views/CallScreen.tsx (hunk 1 above)
Comment (mgamis-msft) [reply samples/CallWithChat/src/app/views/CallScreen.tsx:154]: Yes, I need to hide the screenshare button on IPad. `formFactor` is assigned to desktop on Safari for IPad.

**Review by JamesBurnside — APPROVED** [review]

File: samples/CallWithChat/src/app/views/CallScreen.tsx (hunk 1 above)
Comment (mgamis-msft) [reply samples/CallWithChat/src/app/views/CallScreen.tsx:154]: Also when I set `formFactor` to mobile for IPad. The call control buttons would be compact.

**Review by dmceachernmsft — APPROVED** [review]
